src/Microsoft.NetCore.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -0,0 +1,465 @@
+#pragma warning restore CA1308 // Normalize strings to uppercase
+
+        /// <summary>
+        /// Returns a set of descriptors for the diagnostics that this analyzer is capable of producing.
+        /// </summary>
+        /// <value>The supported diagnostics.</value>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_syncRule, s_asyncRule);
+
+        /// <summary>
+        /// Called once at session start to register actions in the analysis context.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        /// <summary>
+        /// Called on compilation start.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        private static void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            const string SyncMethodName = "Count";
+
+            if (WellKnownTypes.Enumerable(context.Compilation) is INamedTypeSymbol enumerableType)
+            {
+                var operationActionsHandler = new OperationActionsHandler(
+                    targetType: enumerableType,
+                    targetMethod: SyncMethodName,
+                    isAsync: false,
+                    rule: s_syncRule);
+
+                context.RegisterOperationAction(
+                    operationActionsHandler.AnalyzeInvocationOperation,
+                    OperationKind.Invocation);
+
+                context.RegisterOperationAction(
+                    operationActionsHandler.AnalyzeBinaryOperation,
+                    OperationKind.BinaryOperator);
+            }
+
+            if (WellKnownTypes.Queryable(context.Compilation) is INamedTypeSymbol queriableType)
+            {
+                var operationActionsHandler = new OperationActionsHandler(
+                    targetType: queriableType,
+                    targetMethod: SyncMethodName,
+                    isAsync: false,
+                    rule: s_syncRule);
+
+                context.RegisterOperationAction(
+                    operationActionsHandler.AnalyzeInvocationOperation,
+                    OperationKind.Invocation);
+
+                context.RegisterOperationAction(
+                    operationActionsHandler.AnalyzeBinaryOperation,
+                    OperationKind.BinaryOperator);
+            }
+
+            const string AsyncMethodName = "CountAsync";
+
+            if (context.Compilation.GetTypeByMetadataName("Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions") is INamedTypeSymbol entityFrameworkQueryableExtensionsType)
```

Comment:
Should the type be added to `WellKnownTypes`?

Reply:
I thought about it, but it seemed so specific.